src/utils/child_process.rs

```diff
@@ -89,8 +91,135 @@ impl Nanny {
 }
 
 fn boot_desktop_file(path: &Path) -> std::io::Result<Child> {
-    let args = format!( "`if [ \"$(grep '^X-GNOME-Autostart-enabled' {:?} | tail -1 | sed 's/^X-GNOME-Autostart-enabled=//' | tr '[A-Z]' '[a-z]')\" != 'false' ]; then grep '^Exec' {:?} | tail -1 | sed 's/^Exec=//' | sed 's/%.//' | sed 's/^\"//g' | sed 's/\" *$//g'; else echo 'exit'; fi`", path , path);
-    Command::new("sh").arg("-c").arg(args).spawn()
+    use std::io::Error as StdError;
+    let entry = DesktopEntry::parse_file(path)?;
+    let dw_env = std::env::var("XDG_CURRENT_DESKTOP").unwrap_or(Default::default());
```

Comment:
Sorry if I am missing something obvious here, but whats `dw_env` standing for? (The env part is clear to me, btw.)